syft/pkg/cataloger/dotnet/config.go

```diff
@@ -0,0 +1,255 @@
+		if credentialParts := strings.Split(credential, ":"); len(credentialParts) == 2 {
+			g.ProviderCredentials = append(g.ProviderCredentials, nugetProviderCredential{
+				Username: credentialParts[0],
+				Password: credentialParts[1],
+			})
+		}
+	}
+	return g
+}
+
+type sourceAPIResource struct {
+	ID      string `json:"@id"`
+	Type    string `json:"@type"`
+	Comment string `json:"comment,omitempty"`
+}
+
+type sourceAPIContext struct {
+	Vocab   string `json:"@vocab"`
+	Comment string `json:"comment,omitempty"`
+}
+
+type sourceAPI struct {
+	Version   string              `json:"version"`
+	Resources []sourceAPIResource `json:"resources"`
+	Context   *sourceAPIContext   `json:"@context,omitempty"`
+}
+
+func determineDotnetExecutablePath() (string, error) {
+	dotnetPath := ""
+	var err error
+	if runtime.GOOS == "windows" {
+		dotnetPath, err = exec.LookPath("dotnet.exe")
+	} else {
+		dotnetPath, err = exec.LookPath("dotnet")
+	}
+
+	if errors.Is(err, exec.ErrDot) {
+		err = nil
+	}
+
+	return dotnetPath, err
+}
+
+func getPackageSourcesFromSDK(includeDisabledSources bool) []string {
+	// Try to find enabled remote nuget package sources
+	packageSources := []string{}
+
+	if dotnetPath, err := determineDotnetExecutablePath(); err == nil {
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+		defer cancel()
+
+		cmd := exec.CommandContext(ctx, dotnetPath, "nuget", "list", "source", "--format", "Short")
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			return packageSources
+		}
+
+		err = cmd.Start()
+		if err != nil {
+			return packageSources
+		}
+
+		data, err := io.ReadAll(stdout)
+		if err != nil {
+			return packageSources
+		}
+
+		lines := strings.Split(string(data), "\n")
+
+		packageSources = append(packageSources, parseSDKPackageSourcePathsOutput(lines, includeDisabledSources)...)
+	}
+
+	return packageSources
+}
+
+func parseSDKPackageSourcePathsOutput(outputLines []string, includeDisabledSources bool) []string {
+	packageSources := []string{}
+
+	for _, line := range outputLines {
+		line = strings.TrimSpace(line)
+		// Expect something like
+		// E https://api.nuget.org/v3/index.json
+		// or
+		// D https://api.nuget.org/v3/index.json
+
+		// Only enabled sources
+		if strings.HasPrefix(line, "E ") || (strings.HasPrefix(line, "D ") && includeDisabledSources) {
+			packageSource := strings.TrimSpace(line[2:])
+			if strings.HasPrefix(packageSource, "https://") {
+				found := false
+				for _, knownSource := range packageSources {
+					if packageSource == knownSource {
+						found = true
+					}
+				}
+				if !found {
+					packageSources = append(packageSources, packageSource)
+				}
+			}
+		}
+	}
+
+	return packageSources
+}
+
+func getDefaultProviders() string {
+	defaultProvidersMutex.Lock()
+	defer defaultProvidersMutex.Unlock()
+
+	if defaultProviders != "" {
+		return defaultProviders
+	}
+
+	defaultProviders = defaultProvider
+
+	packageSources := getPackageSourcesFromSDK(false)
```

Comment:
If the user doesn't have dotnet installed on the host, it looks like this will fail because it doesn't have a default URL to access for remote data -- is there one or more standard URLs that could just be hardcoded here, at least in the case dotnet doesn't exist?

I ask this because there are a lot of cases where people are scanning something with Syft and they won't have dotnet available on the host. For example: if scanning containers in a small cloud runner; it would be great to be able to resolve licenses in all cases for these if the user enables the feature

Reply:
The default provider is defined in line 17. 😉 

```
const (
	defaultProvider = "https://api.nuget.org/v3-flatcontainer/"
)
```